<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GameWindow</class>
 <widget class="QMainWindow" name="GameWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>826</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>GameWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="codeGroup">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>601</width>
      <height>541</height>
     </rect>
    </property>
    <property name="title">
     <string>Code:</string>
    </property>
    <widget class="CodeEditor" name="codeBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>581</width>
       <height>451</height>
      </rect>
     </property>
     <property name="documentTitle">
      <string/>
     </property>
    </widget>
    <widget class="QPushButton" name="runButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>111</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Run</string>
     </property>
     <property name="icon">
      <iconset resource="resources.qrc">
       <normaloff>:/res/symbols/play-solid.svg</normaloff>:/res/symbols/play-solid.svg</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
     <property name="shortcut">
      <string>F5</string>
     </property>
    </widget>
    <widget class="QPushButton" name="clearButton">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>30</y>
       <width>111</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Clear</string>
     </property>
     <property name="icon">
      <iconset resource="resources.qrc">
       <normaloff>:/res/symbols/spinner-solid.svg</normaloff>:/res/symbols/spinner-solid.svg</iconset>
     </property>
     <property name="iconSize">
      <size>
       <width>24</width>
       <height>24</height>
      </size>
     </property>
     <property name="shortcut">
      <string>F5</string>
     </property>
    </widget>
    <widget class="QSlider" name="ipcSlider">
     <property name="geometry">
      <rect>
       <x>430</x>
       <y>50</y>
       <width>151</width>
       <height>25</height>
      </rect>
     </property>
     <property name="minimum">
      <number>20</number>
     </property>
     <property name="maximum">
      <number>400</number>
     </property>
     <property name="singleStep">
      <number>20</number>
     </property>
     <property name="pageStep">
      <number>1</number>
     </property>
     <property name="value">
      <number>100</number>
     </property>
     <property name="tracking">
      <bool>true</bool>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="tickPosition">
      <enum>QSlider::NoTicks</enum>
     </property>
     <property name="tickInterval">
      <number>20</number>
     </property>
    </widget>
    <widget class="QLineEdit" name="ipcInput">
     <property name="geometry">
      <rect>
       <x>382</x>
       <y>50</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="maxLength">
      <number>3</number>
     </property>
    </widget>
    <widget class="QLabel" name="ipcSpeedLabel">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>30</y>
       <width>161</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Execution Speed (%): </string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="outputGroup">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>570</y>
      <width>601</width>
      <height>221</height>
     </rect>
    </property>
    <property name="title">
     <string>Output:</string>
    </property>
    <widget class="QPlainTextEdit" name="outputBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>581</width>
       <height>181</height>
      </rect>
     </property>
     <property name="textInteractionFlags">
      <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
     </property>
     <property name="centerOnScroll">
      <bool>true</bool>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>20</y>
      <width>611</width>
      <height>771</height>
     </rect>
    </property>
    <property name="title">
     <string>Level Group</string>
    </property>
    <widget class="QLabel" name="levelTitle">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>10</y>
       <width>570</width>
       <height>51</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>28</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Level Title</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QGraphicsView" name="gameDisplay">
     <property name="geometry">
      <rect>
       <x>28</x>
       <y>190</y>
       <width>555</width>
       <height>555</height>
      </rect>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOff</enum>
     </property>
     <property name="sceneRect">
      <rectf>
       <x>0.000000000000000</x>
       <y>0.000000000000000</y>
       <width>550.000000000000000</width>
       <height>550.000000000000000</height>
      </rectf>
     </property>
    </widget>
    <widget class="QLabel" name="levelObjectivesLabel">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>40</y>
       <width>141</width>
       <height>50</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>16</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Objectives:</string>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="levelObjectives">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>86</y>
       <width>160</width>
       <height>93</height>
      </rect>
     </property>
     <property name="textInteractionFlags">
      <set>Qt::NoTextInteraction</set>
     </property>
    </widget>
    <widget class="QTabWidget" name="tabWidget">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>60</y>
       <width>371</width>
       <height>121</height>
      </rect>
     </property>
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Level Info</string>
      </attribute>
      <widget class="QPlainTextEdit" name="levelDescription">
       <property name="geometry">
        <rect>
         <x>-1</x>
         <y>-1</y>
         <width>371</width>
         <height>93</height>
        </rect>
       </property>
       <property name="textInteractionFlags">
        <set>Qt::NoTextInteraction</set>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Lesson</string>
      </attribute>
      <widget class="QPlainTextEdit" name="levelHints">
       <property name="geometry">
        <rect>
         <x>-1</x>
         <y>-1</y>
         <width>371</width>
         <height>93</height>
        </rect>
       </property>
       <property name="textInteractionFlags">
        <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="levelcomplete" name="levelCompleteWidget" native="true">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>-20</y>
      <width>1281</width>
      <height>821</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CodeEditor</class>
   <extends>QPlainTextEdit</extends>
   <header>codeeditor.h</header>
  </customwidget>
  <customwidget>
   <class>levelcomplete</class>
   <extends>QWidget</extends>
   <header location="global">levelcomplete.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
